File: pkg/config/s3.go
package config

// S3Config holds S3/object-storage connection parameters.
type S3Config struct {
	EndpointURL    string
	Bucket         string
	Region         string
	AccessKeyID    string
	SecretAccessKey string
}

// LoadS3 reads S3 connection config from standard env vars.
func LoadS3() S3Config {
	return S3Config{
		EndpointURL:    env("S3_ENDPOINT_URL", "http://localstack:4566"),
		Bucket:         env("S3_BUCKET", "continuo"),
		Region:         env("AWS_DEFAULT_REGION", "us-east-1"),
		AccessKeyID:    env("AWS_ACCESS_KEY_ID", "test"),
		SecretAccessKey: env("AWS_SECRET_ACCESS_KEY", "test"),
	}
}

File: ui/src/server/s3.ts
import { S3Client, GetObjectCommand } from '@aws-sdk/client-s3';

const s3 = new S3Client({
  region: process.env.AWS_DEFAULT_REGION || 'us-east-1',
  endpoint: process.env.S3_ENDPOINT_URL,
  forcePathStyle: true, // Required for MinIO / path-style S3 endpoints
  credentials: {
    accessKeyId: process.env.AWS_ACCESS_KEY_ID || 'test',
    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY || 'test',
  },
});

const BUCKET = process.env.S3_BUCKET || 'continuo';

export async function getLogObject(key: string): Promise<string> {
  const resp = await s3.send(new GetObjectCommand({ Bucket: BUCKET, Key: key }));
  return resp.Body!.transformToString();
}
